### ISSUE TITLE
exited with code [1] via signal [SIGINT]

### ISSUE THREAD

**Issue by garyc2232** [Unitech/pm2#4632]
## What's going wrong?
I have 4 app running over pm2 on linux server (1 GB ram). However, 1 of my app will always restarted due to `exited with code [1] via signal [SIGINT]`. Sometime it can stay for a week without restart but sometime it restart few time in a minute.

Before this issue happened, I was running on pm2 v 2.x and its very stable. And now I have updated to 4.2.3 and still got the issue.

I cant find what is `code [1]` about.  And I have check my logs, it didn't show any error or warning before the restart occur.

> free -h
total        used        free      shared  buff/cache   available
Mem:           957M        353M        149M        728K        455M        411M
Swap:            0B          0B          0B

> PM2        | 2020-03-13T11:11:10: PM2 log: App [app:2] online
PM2        | 2020-03-13T11:11:13: PM2 log: App [app:2] exited with code [1] via signal [SIGINT]
PM2        | 2020-03-13T11:11:13: PM2 log: App [app:2] starting in -fork mode-
PM2        | 2020-03-13T11:11:13: PM2 log: App [app:2] online
PM2        | 2020-03-13T11:11:16: PM2 log: App [app:2] exited with code [1] via signal [SIGINT]
PM2        | 2020-03-13T11:11:16: PM2 log: App [app:2] starting in -fork mode-
PM2        | 2020-03-13T11:11:16: PM2 log: App [app:2] online

**Comment by sataqi** [comment]
facing same issue, were you able to solve?

**Comment by sirajabbas** [comment]
Same issue. Here is the output of my `pm2 log` command 

```
App name:app id:0 disconnected
PM2      | App [app:0] exited with code [1] via signal [SIGINT]
PM2      | App [app:0] starting in -cluster mode-
PM2      | App [app:1] online
PM2      | App name:app id:1 disconnected
PM2      | App [app:1] exited with code [1] via signal [SIGINT]
PM2      | App [app:1] starting in -cluster mode-
PM2      | App [app:0] online
PM2      | App name:app id:0 disconnected
PM2      | App [app:0] exited with code [1] via signal [SIGINT]
PM2      | App [app:0] starting in -cluster mode-
PM2      | App [app:1] online
PM2      | App name:app id:1 disconnected
```

**Comment by sirajabbas** [comment]
Solved the problem 

```
pm2 kill
rm -rf ~/.pm2 
```

**Comment by Ven2575** [comment]
It doesn't solve in my case. Still problem exist!

**Comment by Tashima42** [comment]
> It doesn't solve in my case. Still problem exist!
Same thing here. Node 14.15 and Ubuntu 20.10

**Comment by Macha-orange** [comment]
> > It doesn't solve in my case. Still problem exist!
> 
> Same thing here. Node 14.15 and Ubuntu 20.10
> 
> [edit]
> 
> I was using the wrong file. Everything works fine now.

Hello,

Can you please be more specific how did you fixed it ?

**Comment by Tashima42** [comment]
> > > It doesn't solve in my case. Still problem exist!
> > 
> > 
> > Same thing here. Node 14.15 and Ubuntu 20.10
> > [edit]
> > I was using the wrong file. Everything works fine now.
> 
> Hello,
> 
> Can you please be more specific how did you fixed it ?

Usually on a NodeJs + Express app you would have a `index.js` file with all the main parts. In my case, I have two separate files.

`src/index.js`
```js
import express = from 'express'
const app = express()

app.get('/', (req, res) => {
  res.send('Hello World!')
})

export default app;
```

`src/server.js`
```js
import app from './index.js'
const port = 3000

app.listen(port, () => {
  console.log(`Example app listening at http://localhost:${port}`)
})
```

I was trying to start using the `src/index.js`, which would never work, when I used the correct file, `src/server.js`, everything worked fine.

**Comment by Macha-orange** [comment]
ok thanks

**Comment by llmichaelriveral** [comment]
Still have the same problem 

I have 5 apps running with pm2, i'm trying to run a TypeScript app built with AdonisJS and get ;

PM2      | App [server:0] exited with code [0] via signal [SIGINT]
PM2      | App [server:0] starting in -fork mode-
PM2      | App [server:0] online

15 times and the the app stops with "errored" status

does anybody can help me

**Comment by fantik11** [comment]
> Still have the same problem
> 
> I have 5 apps running with pm2, i'm trying to run a TypeScript app built with AdonisJS and get ;
> 
> PM2 | App [server:0] exited with code [0] via signal [SIGINT] PM2 | App [server:0] starting in -fork mode- PM2 | App [server:0] online
> 
> 15 times and the the app stops with "errored" status
> 
> does anybody can help me

Facing the same error, do you solve it?

**Comment by erkage** [comment]
For me it was Fastify port conflict that caused this error msg.
The core reason that i ended up here is that Fastify logger, which used to print out the error, was turned off by default.
So the conclusion is check all place where your code can broke things and handle and log it.

**Comment by Suhasnama** [comment]
TLDR: Use the js file name as value of script key in ecosystem.config.js , `script : "index.js"` (correct) rather than `script : "node index.js"`(wrong) when using pm2 in cluster mode.

In my case, the issue is caused by wrong configuration of app in ecosystem.config.js for cluster mode.

```
// Wrong config, when in cluster mode
// But works fine when set `exec_mode: "fork"`
{
  name: "api-server",
  script: "npm",
  args: "run start",
  exec_mode: "cluster"
}
```

```
// Correct config, when in cluster mode
{
  name: "api-server",
  script: "index.js",
  exec_mode: "cluster"
}
```

**Comment by asadidebuger** [comment]
This config in ecosystem.config.js file worked successfully for my app:
```
module.exports = {
  apps : [{
    name: "ims",
    script: "npm",
	args: "start"
  }],
  exec_mode: "fork"
}

```

**Comment by aqueeb7** [comment]
It's not working for me folks please help me

**Comment by asadidebuger** [comment]
create `ecosystem.config.js` file in root of project
paste 
```
module.exports = {
  apps : [{
    name: "ims",
    script: "npm",
	args: "start"
  }],
  exec_mode: "fork"
}
```
into it.
if you use npm and your project start from start.js you can use it else you must replace correct `script` and `args` parameters...

**Comment by kharithomas** [comment]
In case it helps anyone in the future:

In my setup, I was facing this issue but couldn't get the _output_ as my Node.js application was intentionally supressing console.log and console.error in the production environment.

After fixing this, I was able to see the output using `docker logs <container-id>` since I have containerized my application (you can use any other method like `pm2 log` or something else to see the application output).

From there, I found my application was failing to launch due to a syntax error in my `package-lock.json` file caused by a merge conflict not caught during the deployment process. 

I removed the `node_modules` folder and ran `npm install` which ultimately fixed my issue.

If you're facing this issue it's highly-likely an **application** issue. To prevent this in the future make sure to add sufficient amounts of console.log and console.error statements to your app or even add a package like Winston to save these logs to a file on your server for better troubleshooting.

Good luck!

**Comment by t-cheeweng** [comment]
In my case , 
i am using 
`npx express-generator --view=ejs bookstore`
to generate my express project , 

i found that the server.listen() / app.listen() is located at app.js
`.bin/www`

so i modify the code, and it works
```
module.exports = {
  apps : [{
    name   : "app1",
    script : "./bin/www", 
    
    env: {
      PORT:3000
    }

  }]
}
```
i think it is depends on which of your file contains  server.listen() / app.listen()  .

**Comment by shikhartechxle** [comment]
I have started facing this error recently, not sure all of sudden getting the same error with pm2 latest version and node. 
I am creating the image in dockerfile running few set of commands. checking the logs I got same error
Can anyone update on this?

**Comment by ynunezaguirre** [comment]
in my case, as a someone already said, i was running the pm2 using a script command (prod) in the package.json  command like this: 

`pm2 start --name back -i max npm -- run prod`

The quick fix for me was to change it and run directly the file

`pm2 start --name back -i max dist/main.js`

**Comment by Solaimanx** [comment]
pm2 kill
rm -rf ~/.pm2  

Worked for me
